### PR TITLE
Add the S50 Keyboard firmware to QMK

### PR DESCRIPTION
**Pull request by joedinkle**
## Description
This PR would add the S50 keyboard, under the eyeohdesigns folder, to QMK.  The S50 keyboard is the first keyboard available under Eye Oh Designs.  It recently finished a group buy, selling approx 55 units.  I want to make sure all my users have access to the firmware.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x ] I have added tests to cover my changes.
- [x ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by astolman** [comment]
s50.c tries to import "S50.h" but the file has a lower case s, it's 's50.h'

**Comment by fauxpark** [comment]
@joedinkle please remove the design files from this PR.

File: keyboards/eyeohdesigns/s50/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+//#include S50_H
+// Defines the keycodes used by our macros in process_record_user
+
+#define ______ KC_TRNS
```
Comment (drashna) [inline keyboards/eyeohdesigns/s50/keymaps/default/keymap.c:20]: These defines aren't necessary, as these are actually defined in the core.  You can safely remove them. 

https://github.com/qmk/qmk_firmware/blob/a0309db983150fd0197eb620a6ba552d90d29f93/quantum/quantum_keycodes.h#L30-L31


```suggestion
```

File: keyboards/eyeohdesigns/s50/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+#define QWERTY 0
+#define RAISE 1
+#define NUMBRS 2
+enum custom_keycodes { QMKBEST = SAFE_RANGE, QMKURL };
```
Comment (drashna) [inline keyboards/eyeohdesigns/s50/keymaps/default/keymap.c:24]: We recommend using enums for the layers, as it's easier to maintain, long term. 

```suggestion
enum layers {
    QWERTY,
    RAISE,
    NUMBRS,
};

enum custom_keycodes { 
    QMKBEST = SAFE_RANGE, 
    QMKURL,
};
```

File: keyboards/eyeohdesigns/s50/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+            break;
+    }
+    return true;
+}
+
+void matrix_init_user(void) {}
+
+void matrix_scan_user(void) {}
+
+void led_set_user(uint8_t usb_led) {}
```
Comment (drashna) [inline keyboards/eyeohdesigns/s50/keymaps/default/keymap.c:116]: These aren't being used and can be removed 

```suggestion
```

File: keyboards/eyeohdesigns/s50/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+
+void encoder_update_user(uint8_t index, bool clockwise) {
+ if (index == 0) {
+	 if (clockwise) {
+  tap_code(KC_VOLU);
+ }  else{
+	tap_code (KC_VOLD);
+ }
+ }
+}
```
Comment (drashna) [inline keyboards/eyeohdesigns/s50/keymaps/default/keymap.c:126]: Formatting:
```suggestion
void encoder_update_user(uint8_t index, bool clockwise) {
    if (index == 0) {
        if (clockwise) {
            tap_code(KC_VOLU);
        } else {
            tap_code (KC_VOLD);
        }
     }
}
```

File: keyboards/eyeohdesigns/s50/s50.h [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+//#pragma once
+#ifndef S50_H
+#define S50_H
```
Comment (drashna) [inline keyboards/eyeohdesigns/s50/s50.h:18]:
```suggestion
#pragma once

```

File: keyboards/eyeohdesigns/s50/s50.h [hunk 6]
```diff
@@ -0,0 +1,238 @@
...
+
+/* This a shortcut to help you visually see your layout.
+ *
+ * The first section contains all of the arguments representing the physical
+ * layout of the board and position of the keys.
+ *
+ * The second converts the arguments into a two-dimensional array which
+ * represents the switch matrix.
+ */
+#define ANSI_Left( \
```
Comment (drashna) [inline keyboards/eyeohdesigns/s50/s50.h:29]: For the layout macros, we have them as `LAYOUT_*`. and the rest is all lower case. 

```suggestion
#define LAYOUT_ansi_left( \
```

**Comment by joedinkle** [comment]
Changes are made in another branch.  Closing this PR and will resubmit in another branch.

**Pull request closed without merging**